comment/src/lib.rs

```diff
@@ -0,0 +1,124 @@
+    version: env!("CARGO_PKG_VERSION"),
+    usage: r#"
+Usage
+
+    rad comment <id> [--description <text>] [--reply-to <index>]
+
+Options
+
+    --description <text>    Comment text
+    --reply-to <index>      Index of comment writing a reply for
+    --help                  Print help
+"#,
+};
+
+#[derive(Debug)]
+pub struct Options {
+    pub id: CobIdentifier,
+    pub description: Option<String>,
+    pub reply_index: Option<CommentId>,
+}
+
+impl Args for Options {
+    fn from_args(args: Vec<OsString>) -> anyhow::Result<(Self, Vec<OsString>)> {
+        use lexopt::prelude::*;
+
+        let mut parser = lexopt::Parser::from_args(args);
+        let mut id: Option<CobIdentifier> = None;
+        let mut description: Option<String> = None;
+        let mut reply_index: Option<CommentId> = None;
+
+        while let Some(arg) = parser.next()? {
+            match arg {
+                Long("help") => {
+                    return Err(Error::Help.into());
+                }
+                Long("description") => {
+                    description = Some(parser.value()?.to_string_lossy().into());
+                }
+                Long("reply-to") => {
+                    let idx = parser
+                        .value()?
+                        .parse::<usize>()
+                        .map_err(|_| anyhow!("index for `--reply-to` can't be parsed as usize"))?;
+
+                    reply_index = Some(CommentId::from(idx));
+                }
+                Value(val) if id.is_none() => {
+                    let val = val
+                        .to_str()
+                        .ok_or_else(|| anyhow!("issue id specified is not UTF-8"))?;
+
+                    id = Some(
+                        CobIdentifier::from_str(val)
+                            .map_err(|_| anyhow!("invalid issue id '{}'", val))?,
+                    );
+                }
+                _ => {
+                    return Err(anyhow!(arg.unexpected()));
+                }
+            }
+        }
+
+        Ok((
+            Options {
+                id: id.ok_or_else(|| anyhow!("an issue id must be provided"))?,
+                description,
+                reply_index,
+            },
+            vec![],
+        ))
+    }
+}
+
+pub fn run(options: Options) -> anyhow::Result<()> {
+    let profile = profile::default()?;
+    let signer = term::signer(&profile)?;
+    let storage = keys::storage(&profile, signer)?;
+    let (project, _) = project::cwd()?;
+    let whoami = person::local(&storage)?;
+    let issues = Issues::new(whoami.clone(), profile.paths(), &storage)?;
+    let patches = cobs::patch::Patches::new(whoami, profile.paths(), &storage)?;
+    let cob_id = options.id;
+
+    let doc = options
+        .description
+        .unwrap_or("Enter a description...".to_owned());
+
+    if let Some(text) = term::Editor::new().edit(&doc)? {
+        if let Ok(id) = issues.resolve_id(&project, cob_id.clone()) {
```

Comment:
Keep in mind, with current implementation of `resolve_id` this would always return `Ok(...)` if `cob_id` is of the full variant. I've omitted my `.find(...)` solution since you were not happy, and we need to find another fix for this.